File: backend/src/main/java/project/backend/global/security/app/CookieUtils.java
package project.backend.global.security.app;

import jakarta.servlet.http.Cookie;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.util.Arrays;
import java.util.Optional;

public class CookieUtils {

	public static void saveCookie(HttpServletResponse response, String accessToken) {
		Cookie cookie = new Cookie("accessToken", accessToken);
		cookie.setPath("/");
		cookie.setHttpOnly(true);
		cookie.setMaxAge(60 * 10);
		response.addCookie(cookie);
	}

	public static void deleteCookie(HttpServletResponse response) {
		Cookie cookie = new Cookie("accessToken", null);
		cookie.setPath("/");
		cookie.setHttpOnly(true);
		cookie.setMaxAge(0);
		response.addCookie(cookie);
	}

	public static Optional<Cookie> getCookie(HttpServletRequest request, String name) {
		Cookie[] cookies = request.getCookies();
		if (cookies == null) {
			return Optional.empty();
		}

		return Arrays.stream(cookies)
			.filter(c -> c.getName().equals(name))
			.findFirst();
	}

}

File: backend/src/main/java/project/backend/auth/app/CookieUtils.kt
package project.backend.auth.app

import jakarta.servlet.http.Cookie
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse

object CookieUtils {

    fun saveCookie(response: HttpServletResponse, accessToken: String) {
        val cookie = Cookie("accessToken", accessToken).apply {
            path = "/"
            isHttpOnly = true
            maxAge = 60 * 60 * 24 * 7
        }
        response.addCookie(cookie)
    }

    fun deleteCookie(response: HttpServletResponse) {
        val cookie = Cookie("accessToken", null).apply {
            path = "/"
            isHttpOnly = true
            maxAge = 0
        }
        response.addCookie(cookie)
    }

    fun getCookie(request: HttpServletRequest, name: String): Cookie? {
        return request.cookies
            ?.find { it.name == name }
    }
}